Compile failed on my MacM2Max

## Bug Report

<!-- Thanks for your bug report! Don't worry if you can't fill out all the sections. -->

### What version of TiKV are you using?
I built TiKV from source code with tag release-8.5.

### What operating system and CPU are you using?
MacOS 14.7.1 (23H222)

### Steps to reproduce
just execute `make` command.

### What did you expect?
build success.

### What did happened?
build failed with these outputs.
It seems build failed because of a macro in ZLIB dependency.
cargo:warning=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/_stdio.h:314:7: error: expected ')'

  cargo:warning=  314 | FILE    *fdopen(int, const char *) __DARWIN_ALIAS_STARTING(__MAC_10_6, __IPHONE_2_0, __DARWIN_ALIAS(fdopen));

  cargo:warning=      |          ^

  cargo:warning=src/zlib/zutil.h:140:33: note: expanded from macro 'fdopen'

  cargo:warning=  140 | #        define fdopen(fd,mode) NULL /* No fdopen() */

  cargo:warning=      |                                 ^

  cargo:warning=/opt/homebrew/Cellar/llvm/19.1.3/lib/clang/19/include/__stddef_null.h:26:22: note: expanded from macro 'NULL'

  cargo:warning=   26 | #define NULL ((void*)0)

  cargo:warning=      |                      ^

  cargo:warning=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/_stdio.h:314:7: note: to match this '('

  cargo:warning=src/zlib/zutil.h:140:33: note: expanded from macro 'fdopen'

  cargo:warning=  140 | #        define fdopen(fd,mode) NULL /* No fdopen() */

  cargo:warning=      |                                 ^

  cargo:warning=/opt/homebrew/Cellar/llvm/19.1.3/lib/clang/19/include/__stddef_null.h:26:14: note: expanded from macro 'NULL'

  cargo:warning=   26 | #define NULL ((void*)0)

  cargo:warning=      |              ^

  cargo:warning=21 warnings generated.

--- a/Makefile
+++ b/Makefile
@@ -56,6 +56,13 @@ export CXXFLAGS := $(CXXFLAGS) -fno-omit-frame-pointer -mno-omit-leaf-frame-poin
 ENABLE_FEATURES += pprof-fp
 endif
 
+# This is a workaround for newer versions of Clang (used by Xcode/macOS) that treat variable-length arrays (VLAs) as error.
+# VLAs are used by RocksDB's encryption code.
+ifeq ($(shell uname -s),Darwin)
+export CFLAGS := $(CFLAGS) -Wno-error=vla -Wno-vla-extension
+export CXXFLAGS := $(CXXFLAGS) -Wno-error=vla -Wno-vla-extension
+endif
+
 # Pick an allocator
 ifeq ($(TCMALLOC),1)
 ENABLE_FEATURES += tcmalloc

